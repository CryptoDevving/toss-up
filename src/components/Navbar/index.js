import dice_logo from "../../assets/dice_logo.png";
import React, { Component } from "react";

class Navbar extends Component {
  render() {
    return (
      <nav className="navbar fixed-top  p-0 shadow text-monospace">
        <a
          className="navbar-brand col-sm-3"
          href="http://brassey.io/"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img src={dice_logo} height="40" alt="logo" /> TossUp
        </a>
        <div className="hand">
          {/* Wallet Connection */}
          <ul className="navbar-nav">
            {!this.props.account && !this.props.loading ? (
              <div className="row text-center text-monospace">
                <button
                  type="submit"
                  onClick={(e) => this.props.on(e)}
                  className="btn high btn-sm"
                  style={{ width: "125px", fontSize: "17px" }}
                >
                  <b>Connect</b>
                </button>
                &nbsp;
              </div>
            ) : !this.props.account && this.props.loading ? (
              <div className="row text-center text-monospace">
                <button
                  type="submit"
                  className="btn high btn-sm"
                  style={{ width: "125px", fontSize: "17px" }}
                  disabled
                >
                  <span
                    className="spinner-border spinner-border-sm"
                    role="status"
                    aria-hidden="true"
                  ></span>
                  <span className="sr-only">Loading...</span>
                </button>
                &nbsp;
              </div>
            ) : (
              <div className="row text-center text-monospace">
                <button
                  type="submit"
                  onClick={(e) => this.props.off(e)}
                  className="btn low btn-sm"
                  style={{ width: "125px", fontSize: "17px" }}
                >
                  Disconnect
                </button>
                &nbsp;
              </div>
            )}
          </ul>
        </div>

        {/* ETH Address */}
        {!this.props.account ? (
          <div
            id="loader"
            className="spinner-border"
            role="status"
          ></div>
        ) : (
          <li className="nav-item text-nowrap d-sm-block address">
            <a
              className="text-light address"
              href={
                "https://rinkeby.etherscan.io/address/" + this.props.account
              }
              target="_blank"
              rel="noopener noreferrer"
            >
              {this.props.account}
            </a>{" "}
            &nbsp;
          </li>
        )}
      </nav>
    );
  }
}

export default Navbar;
