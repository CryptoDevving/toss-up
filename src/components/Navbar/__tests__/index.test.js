import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';

import Navbar from '..';

afterEach(cleanup);

describe('Navbar component', () => {
  it('renders', () => {
    render(<Navbar />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Navbar />);

    expect(asFragment()).toMatchSnapshot();
  });

})