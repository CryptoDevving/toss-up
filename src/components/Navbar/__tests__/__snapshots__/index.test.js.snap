// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar fixed-top flex-md-nowrap p-0 shadow text-monospace"
  >
    <a
      class="navbar-brand col-sm-3"
      href="http://brassey.io/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="logo"
        height="32"
        src="dice_logo.png"
      />
       TossUp
    </a>
    <div
      class="spinner-border text-light"
      id="loader"
      role="status"
    />
  </nav>
</DocumentFragment>
`;
